src/components/Breadcrumb/index.js

```diff
@@ -14,42 +14,38 @@ import StyledBreadcrumb from './styles/StyledBreadcrumb'
 /**
  * A breadcrumb component.
  */
-class Breadcrumb extends PureComponent {
-  renderElement = element => {
-    const { renderElement } = this.props
-
-    return chain(
-      // We assume a valid result as soon as we got a non null element
-      element => !isNull(element) && !isUndefined(element),
-      // Most precise, we got a custom render function for THAT element
-      () =>
-        element && element.render && isFunction(element.render) ? element.render(element) : null,
-      // A little less precise, we got a custom render function for each item
-      () => (renderElement && isFunction(renderElement) ? renderElement(element) : null),
-      // Least precise, neither other function returned something, we default to some hand made rendering
-      () => (
-        <React.Fragment>
-          {element.url && <BreadcrumbElement element={element} />}
-          {!element.url && <CurrentElement>{element.name}</CurrentElement>}
-        </React.Fragment>
+const Breadcrumb = ({ className, elements, renderElement, separator }) => {
+  const displayElement = useCallback(
+    element =>
+      chain(
+        // We assume a valid result as soon as we got a non null element
+        element => !isNull(element) && !isUndefined(element),
+        // Most precise, we got a custom render function for THAT element
+        () =>
+          element && element.render && isFunction(element.render) ? element.render(element) : null,
+        // A little less precise, we got a custom render function for each item
+        () => (renderElement && isFunction(renderElement) ? renderElement(element) : null),
+        // Least precise, neither other function returned something, we default to some hand made rendering
+        () => (
+          <React.Fragment>
+            {element.url && <BreadcrumbElement element={element} />}
+            {!element.url && <CurrentElement>{element.name}</CurrentElement>}
+          </React.Fragment>
+        ),
       ),
-    )
-  }
+    [renderElement],
+  )
```

Comment:
I'm not sure about the utility of useCallback here. If `displayElement` can keep the same reference between each render, then, what will change for the component ?
I think here, if you want to optimize the component, you need useMemo, to memoize the element.
Something like this :
```
const Element = ({ displayElement, element, separator, withSeparator }) => {
  const elementDisplayed = useMemo(() => displayElement(element), [displayElement, element])

  return <>
          {elementDisplayed}
          {withSeparator  && <BreadcrumbSeparator>{separator}</BreadcrumbSeparator>}
  </ >
}
```
And then you can keep the useCallback if you want to React.memo this new component.
But, for a Breadcrumb, do we really need to work on optimization yet ?

Reply:
👍 You're right, no real need for an optimization here.